components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/guide/guide.js

```diff
@@ -0,0 +1,888 @@
+                    'click #sampleContent': 'Here are some samples for you to try out!',
+                    'showSkip': false,
+                    'showNext': true
+                },
+                {
+                    'click .more-samples': 'Click here to view the full sample list',
+                    'showSkip': false,
+                    'showFalse': false,
+                    'shape': 'rect',
+                    'right': 500
+                },
+                {
+                    'next #sampleDialog' : 'You can try out more samples from here. click <b class="lime-text">Next</b>',
+                    'showSkip' : false,
+                    'scrollAnimationSpeed' : 900,
+                    'bottom' : 150
+                },
+                {
+                    'click #DataPreprocessing' : '<b class="lime-text">Click</b> here to try a sample which ' +
+                        '<b class="lime-text">collect data via TCP transport and preprocess.</b>',
+                    'showNext' : false,
+                    'showSkip' : false,
+                    onBeforeStart: function () {
+                        $('#DataPreprocessing').focus();
+                    }
+                }
+            ];
+        };
+
+        //Constructor for the guide
+        Guide.prototype.constructor = Guide;
+
+        //function to view the dialog box to choose the guide option
+        Guide.prototype.showModel =  function(){
+            this.guideDialog.modal('show').on('shown.bs.modal', function(){
+                $('#guideDialog').trigger('loaded');
+            });
+            this.guideDialog.on('hidden.bs.modal', function(){
+                $('#guideDialog').trigger('unloaded');
+            })
+        };
+
+        //start function for the complete guide
+        Guide.prototype.start = function () {
+
+            var self = this;
+            var guideModal = self.guideDialog.filter("#guideDialog");
+            var hintInstance = new EnjoyHint({});
+            var callback = function() { guideModal.modal('hide') };
+
+            //check whether there are multiple tabs and if the current tab is "welcome-page"
+            _.each(self.tabList, function (tab) {
+                if (self.tabList.length === 1 && tab._title === "welcome-page") {
+                    self.showModel();
+                } else {
+                    DesignViewUtils.prototype.errorAlert(Constants.ERROR_TEXT);
+                }
+            });
+
+            //function for the complete tour
+            guideModal.find("button").filter("#fullGuide").click(function() {
+
+                //unbinding the click events from the previous click
+                $('#fullGuide').off('click');
+                $('#simulationGuide').off('click');
+                $('#sampleGuide').off('click');
+
+                hintInstance.setScript(self.completeGuide);
+                hintInstance.runScript();
+
+                instance = hintInstance;
```

Comment:
Why assigning `hintInstance` to `instance`? Can't we use single variable for this?

Reply:
Changed to use a single variable